packages/arb-validator-core/gobridge/challengeFactory.go

```diff
@@ -23,28 +23,64 @@ import (
 )
 
 type challengeFactory struct {
-	contract common.Address
-	client   *GoArbAuthClient
+	challengeFactoryContract common.Address
+	client                   *GoArbAuthClient
+}
+
+func deployChallengeFactory(m *goEthdata) {
+	m.challenges = make(map[common.Address]*challenge)
+	m.challengeFactoryContract = &challengeFactory{
+		challengeFactoryContract: m.getNextAddress(),
+		client:                   nil,
+	}
 }
 
 func newChallengeFactory(address common.Address, client *GoArbAuthClient) (*challengeFactory, error) {
-	return &challengeFactory{address, client}, nil
+	client.challengeFactoryContract.client = client
+	return client.challengeFactoryContract, nil
 }
```

Comment:
This function is confusing since the `address` parameter is never used

Reply:
cleaned up deployment and binding to contracts